Change page information structure to match the gecko side

canova opened the pull request:
It changes the the page structure of page information array. And adds two upgraders for gecko/processed profiles. Currently we don't do any special handling for single tab browsing yet.

Gecko sides of this patch haven't landed yet, we should first land this side and then we can land the gecko side.
Gecko bugs:
https://bugzilla.mozilla.org/show_bug.cgi?id=1583271
https://bugzilla.mozilla.org/show_bug.cgi?id=1512500

(I will put some profiles here to test with deploy preview)

gregtatum commented on src/types/markers.js:
```diff
@@ -413,8 +413,7 @@ export type DOMEventMarkerPayload = {|
   phase: 0 | 1 | 2 | 3,
-  docShellId?: string,
-  docshellHistoryId?: number,
+  innerWindowID?: number,
```
Do we have a good place to document this? I'm guessing it'll be repeated across multiple payloads.

gregtatum replied:
It's probably good enough to just grep and find the Page object.

gregtatum commented on src/profile-logic/gecko-profile-versioning.js:
```diff
@@ -710,5 +710,98 @@ const _upgraders = {
+          // This information is new. We had isSubFrame field but that's not
+          // useful for us to determine the embedders. Therefore setting older
+          // pages to 0 which means null.
+          page.embedderInnerWindowID = 0;
```
Maybe note that 0 === null on the type for Pages.

gregtatum commented on src/types/markers.js:
```diff
@@ -413,8 +413,7 @@ export type DOMEventMarkerPayload = {|
   eventType: string,
   phase: 0 | 1 | 2 | 3,
-  docShellId?: string,
-  docshellHistoryId?: number,
```
Proper camel casing now! Yay. This was bothering me but not worth fixing.

canova commented:
@gregtatum It's ready for a review again. Changed the json files to test this upgrader. Second commit is the same commit that you reviewed. Added the old page information to the first commit and on the last commit updated the snapshots.
